### PR TITLE
Log4j config

### PR DESCRIPTION
**Pull request by JumaX**
# Description

Add the ability to configure log4j settings for each host.  Specifically, defaulted to rolling log file appender, with the ability to set the number of log files to keep, the size of each log file, and the name and path they should be written under.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:

Have tested on both RHEL 7.x and Ubuntu 18.04.  Full cluster deployment with logging settings adjusted.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.variables/defaults/main.yml [hunk 1]
```diff
@@ -20,6 +20,166 @@ kafka_rest_jolokia_port: 7775
...
+zookeeper_log_name: zookeeper-server.log
+zookeeper_max_log_files: 10
+zookeeper_log_file_size: 100MB
+
+## Garbage Collection logging
+
+zookeeper_gclog_path: /var/log/kafka/
+zookeeper_gclog_name: gc.log
+zookeeper_gclog_max_log_files: 10
+Zookeeper_gclog_log_file_size: 100M
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml]: did you mean for this to be capitalized

File: roles/confluent.variables/defaults/main.yml [hunk 2]
```diff
@@ -30,7 +190,7 @@ jmxexporter_jar_path: /opt/prometheus/jmx_prometheus_javaagent.jar
 kafka_broker_jmxexporter_port: 8080
 
 rbac_enabled: false
-confluent_server_enabled: "{{ true if rbac_enabled|bool else false }}"
+confluent_server_enable_: "{{ true if rbac_enabled|bool else false }}"
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml]: lost the d?

File: roles/confluent.zookeeper/defaults/main.yml [hunk 3]
```diff
@@ -12,6 +12,8 @@ zookeeper_kafka_opts_buildout: "{{ zookeeper_jolokia_java_arg + ' ' + zookeeper_
 zookeeper_service_environment_overrides:
   KAFKA_HEAP_OPTS: "-Xmx1g"
   KAFKA_OPTS: "{{ zookeeper_kafka_opts_buildout }}"
+  KAFKA_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/kafka/zookeeper_log4j.properties"
```
Comment (domenicbove) [inline roles/confluent.zookeeper/defaults/main.yml]: KAFKA_LOG4J_OPTS: "-Dlog4j.configuration=file:{{zookeeper.log4j_file}}"

File: roles/confluent.variables/defaults/main.yml [hunk 4]
```diff
@@ -20,6 +20,166 @@ kafka_rest_jolokia_port: 7775
...
+rest_proxy_appender_log_file_size: 100MB
+
+## Garbage Collection logging
+
+rest_proxy_gclog_path: /var/log/kafka-rest/
+rest_proxy_gclog_name: rest-proxy-gc.log
+rest_proxy_gclog_max_log_files: 10
+rest_proxy_gclog_log_file_size: 100M
+
+## Control Center logging configuration
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml]: move component vars into component/defaults/main.yml if they do not need to be global

**Comment by JumaX** [comment]
@domenicbove updated with requested changes, please review.

File: roles/confluent.variables/defaults/main.yml (hunk 2 above)
Comment (JumaX) [reply roles/confluent.variables/defaults/main.yml]: Fixed.

**Comment by JumaX** [comment]
@domenicbove good catch.  Corrected.

File: roles/confluent.kafka_rest/tasks/main.yml [hunk 5]
```diff
@@ -33,6 +33,14 @@
     recurse: yes
   when: sasl_protocol == 'kerberos'
 
+- name: Create Logs Directory
+  file:
+    path: "{{kafka_rest.appender_log_path}}"
+    owner: "{{kafka_rest.user}}"
+    group: "{{kafka_rest.group}}"
+    mode: '764'
```
Comment (domenicbove) [inline roles/confluent.kafka_rest/tasks/main.yml]: should we keep this task right above the create kafka rest log4j config task?
thats how it is in the other main.ymls

**Pull request merged**